modules/json/src/smithy4s/json/internals/JsonPayloadCodecCompilerImpl.scala

```diff
@@ -53,13 +53,25 @@ private[json] case class JsonPayloadCodecCompilerImpl(
   def fromSchema[A](schema: Schema[A], cache: Cache): PayloadCodec[A] = {
     val jcodec = jsoniterCodecCompiler.fromSchema(schema, cache)
     val reader: PayloadReader[A] = new JsonPayloadReader(jcodec)
-    val writer: PayloadWriter[A] = Writer.encodeBy { (value: A) =>
-      val intermediate = Blob(
-        writeToArray(value, jsoniterWriterConfig)(jcodec)
-      )
-      if (intermediate.sameBytesAs(Blob("null"))) Blob("{}")
-      else intermediate
-    }
+    val writer: PayloadWriter[A] =
+      schema.hints.get(smithy.api.HttpPayload) match {
```

Comment:
perhaps move this match statement inside the Writer to just indicate the empty body as seems like the codec and writer are the same